src/include/detail/linalg/matrix_with_ids.h

```diff
@@ -0,0 +1,266 @@
+/**
+ * @file   matrix.h
```

Comment:
All comments in PR #235 apply.

Reply:
Thanks, though please note that this PR was a draft and so not ready for review yet.